src/krbd.cc

```diff
@@ -406,7 +406,8 @@ static int do_map(struct udev *udev, const krbd_spec& spec, const string& buf,
     goto out_mon;
   }
 
-  mapper = make_named_thread("mapper", [&buf, sysfs_r_fd = fds[1]]() {
```

Comment:
Why not fix the race in `make_named_thread`?

@adamemerson

Reply:
@dillaman I'm submitting a separate PR to removing it entirely.  It's looks attractive but it can't work reliably.

Reply:
I included "krbd: don't use make_named_thread()" commit in this PR instead of the future "remove make_named_thread()" PR because the test case in this PR is triggering this race pretty consistently (and also because it will get backported along with the netlink ENOBUFS series in one go).

Reply:
How about https://github.com/ceph/ceph/pull/31057 ?

Reply:
Ah, I didn't consider std::bind().  I think it would work, although it does add an extra copy/move for the captures.

Reply:
Got rid of the std::bind. Since std::thread does something similar to bind anyway we can just forward arguments to it. This way we don't have two binds composed together.